### PR TITLE
Feature: Dark Theme Integration for the Extension

### PR DESCRIPTION
**Pull request by Thunder-Blaze**
## Proposed Changes
Added a Dark Theme for those who work from the Shadows of World
- Added Apropriate CSS Variable for Dark Theme
- Also added TS for switching between themes and saving the theme value to storage

## Related Issue
- None

## Preview

https://github.com/user-attachments/assets/f3df5aeb-005c-4771-92b8-29ccdfa6d659

## Checklist
- [x] I have tested my changes thoroughly.
- [ ] I have updated the documentation accordingly.
- [x] My code follows the project's coding conventions and style.
- [x] I have added necessary comments to my code, particularly in complex areas.
- [x] I have ensured that my changes do not introduce any new issues.

### CODE REVIEW

**Comment by Thunder-Blaze** [comment]
Also, It provided one of the Paragraph Answers in markdown format

image

**Comment by rootCircle** [comment]
@Gyan172004

File: src/utils/defaultProperties.ts [hunk 1]
```diff
@@ -24,6 +25,7 @@ const DEFAULT_PROPERTIES: typeDefaultProperties = {
   sleep_duration: 1500,
   model: LLMEngineType.Gemini,
   enableConsensus: false,
+  enableDarkTheme: false,
```
Comment (rootCircle) [inline src/utils/defaultProperties.ts]: Set it to true by default, users will obviously love soothing eyes!

File: package.json [hunk 2]
```diff
@@ -79,5 +79,8 @@
     "prettier": "^3.4.2",
     "typescript-eslint": "^8.17.0",
     "web-ext": "^8.3.0"
-  }
+  },
+  "trustedDependencies": [
```
Comment (rootCircle) [inline package.json]: Why this?

**Review by rootCircle — CHANGES_REQUESTED** [review]

**Comment by rootCircle** [comment]
> Also, It provided one of the Paragraph Answers in markdown format
> 
> image
> 

Related to #34! It's better to keep segregated from the original topic!

Thanks for the catch tho! Really appreciated

File: package.json (hunk 2 above)
Comment (Thunder-Blaze) [reply package.json]: it got added by itself when I ran `bun i`
Should I remove ?

File: src/utils/defaultProperties.ts (hunk 1 above)
Comment (Thunder-Blaze) [reply src/utils/defaultProperties.ts]: Sure 👍🏻

**Comment by rootCircle** [comment]
Have reviewed your code, looks all good to me! Can you confirm responsiveness in mobile devices wrt dark theme?

File: package.json (hunk 2 above)
Comment (rootCircle) [reply package.json]: For trusted dependencies right? Let me check once!

**Comment by Thunder-Blaze** [comment]
I haven't changed any code related to responsiveness, only modified colors in the css files
Still checked the page, it is responsive but some parts are overflowing their containers (if you want I can fix this in another or maybe this PR)

image

on another note, I haven't changed color of this warning, should I change it too or let it be ?

image

File: package.json (hunk 2 above)
Comment (Thunder-Blaze) [reply package.json]: yes trusted dependencies, lmk once you check it

Comment (Thunder-Blaze) [reply package.json]: umm any update ?

**Comment by rootCircle** [comment]
> I haven't changed any code related to responsiveness, only modified colors in the css files
Still checked the page, it is responsive but some parts are overflowing their containers (if you want I can fix this in another or maybe this PR)

Yes sure go ahead if you have the bandwidth!

> on another note, I haven't changed color of this warning, should I change it too or let it be ?

It's fine

File: package.json (hunk 2 above)
Comment (rootCircle) [reply package.json]: AFK! Can't really confirm at this moment, but if it was automatically generated, revert those changes, I'll check more on it later!

Comment (Thunder-Blaze) [reply package.json]: sure

**Review by rootCircle — APPROVED** [review]

**Comment by rootCircle** [comment]
Looks good! Thanks for your contributions, you'll be credited as part of future release!

**Pull request merged**